catroid/src/androidTest/java/org/catrobat/catroid/test/cloudmessaging/CloudMessagingTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.cloudmessaging;
+
+import android.test.AndroidTestCase;
+
+import org.catrobat.catroid.cloudmessaging.CloudMessage;
+
+public class CloudMessagingTest extends AndroidTestCase {
+
+	private CloudMessage cloudMessage;
+
+	@Override
+	protected void setUp() throws Exception {
+		super.setUp();
+		cloudMessage = new CloudMessage();
+	}
+
+	public void testAllRequiredFieldsAvailable() {
+
+		assertFalse(cloudMessage.isValidData());
+
+		cloudMessage.setNotificationTitle("Test Notification Title");
+		cloudMessage.setNotificationMessage("Test Notification Message");
+		cloudMessage.setNotificationUrl("https://www.catrobat.org/");
+
+		assertTrue(cloudMessage.isValidData());
+	}
+
+	public void testRequiredFieldIsMissing() {
+
+		assertFalse(cloudMessage.isValidData());
+
+		cloudMessage.setNotificationTitle("Test Notification Title");
+
+		assertFalse(cloudMessage.isValidData());
+
+		cloudMessage.setNotificationMessage("Test Notification Message");
+		cloudMessage.setNotificationUrl("https://www.catrobat.org/");
+
+		assertTrue(cloudMessage.isValidData());
```

Comment:
Actually, you don't need to "fix" the cloud message. The test just needs to test that the CloudMessage is invalid if required fields are missing. Please adapt everywhere.